File: app/channels/note_channel.rb
class NoteChannel < ApplicationCable::Channel
  def subscribed
    stream_from "note_channel_#{params['note']}" 
  end

  def unsubscribed
  end

  def speak(data)
    Message.create! content: data['message'], user_id: current_user.id, note_id: params['note']
  end
end

File: app/javascript/channels/note_channel.js
import consumer from "./consumer"

$(function() {
  const chatChannel = consumer.subscriptions.create({ channel: 'NoteChannel', note: $('#messages').data('note_id') }, {
    connected() {
      // Called when the subscription is ready for use on the server
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received: function(data) {
      return $('#messages').prepend(data['message']);
    },

    speak: function(message) {
      return this.perform('speak', {
        message: message
      });
    }
  });

  $(document).on('keypress', '[data-behavior~=note_speaker]', function(event) {
    if (event.keyCode === 13) {
      chatChannel.speak(event.target.value);
      event.target.value = '';
      return event.preventDefault();
    }
  });
});